### ISSUE TITLE
memoryview obj typing incorrect and memoryview missing some methods/attributes

### ISSUE THREAD

**Issue by Road-hog123** [microsoft/pyright#318]
**Describe the bug**
The `obj` argument of `memoryview(obj)` is of type `Union[bytes, bytearray, memoryview]`, which excludes for example `array.array` which used several times in the examples in the [documentation for memoryview](https://docs.python.org/3/library/stdtypes.html#memoryview).
Additionally, the documentation includes the methods `toreadonly()`, `release()` and `cast()`, and the attribute `obj`, which are all unknown to Pyright.

**To Reproduce**
Save the code below to a file with VSCode and observe the two errors shown when passing `a` into `memoryview()` on line 3, and when attempting to call the `cast()` method of `x` on line 8.

**Expected behavior**
The type of `obj` should include all types in the standard library that support the buffer protocol.
All documented methods and attributes of `memoryview` should be known to Pyright.

**Screenshots or Code**
This code is a copy of the first example in the [documentation for the `cast()` method](https://docs.python.org/3/library/stdtypes.html#memoryview.cast).
```python
import array
a = array.array('l', [1,2,3])
x = memoryview(a)
print(x.format)
print(x.itemsize)
print(len(x))
print(x.nbytes)
y = x.cast('B')
print(y.format)
print(y.itemsize)
print(len(y))
print(y.nbytes)
```

**VS Code extension or command-line**
VS Code extension version 1.0.81

**Comment by erictraut** [comment]
Thanks for the bug report.

Pyright relies on standard type stub libraries from the [python.typeshed](https://github.com/python/typeshed) project. Pyright ships with a recent copy of these type stubs, including [builtins.pyi](https://github.com/python/typeshed/blob/master/stdlib/2and3/builtins.pyi), which defines the interface to the memoryview class.

I've updated the copy of builtins.pyi that I include in pyright to include the missing methods and attribute, and I'll work with the typeshed folks to get these pushed back to that project.

I wasn't able to address the first problem you reported — the error reported on the line `x = memoryview(a)`. The memoryview `__init__` method takes a parameter that is annotated as `obj: Union[bytes, bytearray, memoryview]`. You are attempting to pass a value of type `array[int]`, which isn't assignable to the annotated type. You can fix the issue in your code by changing it `x = memoryview(a.tostring())` or `x = memoryview(a.tobytes())`.

**Comment by erictraut** [comment]
This is now fixed in 1.0.82, which I just published.

**Comment by Road-hog123** [comment]
> I wasn't able to address the first problem you reported — the error reported on the line `x = memoryview(a)`. The memoryview `__init__` method takes a parameter that is annotated as `obj: Union[bytes, bytearray, memoryview]`. You are attempting to pass a value of type `array[int]`, which isn't assignable to the annotated type. You can fix the issue in your code by changing it `x = memoryview(a.tostring())` or `x = memoryview(a.tobytes())`.

On further investigation this is also an issue with typeshed in that the annotation of the `__init__` method doesn't match the types that are passed into it in the official documentation. I've [opened a new issue](https://github.com/python/typeshed/issues/3433) on the typeshed issue tracker as I don't have a list of objects supporting the buffer protocol with which to fix the annotation properly.
